src/proxy/outbound.rs

```diff
@@ -200,14 +200,22 @@ impl OutboundConnection {
                 let code = response.status();
                 match hyper::upgrade::on(response).await {
                     Ok(mut upgraded) => {
-                        super::copy_hbone("hbone client", &mut upgraded, &mut stream)
-                            .await
-                            .expect("hbone client copy");
+                        match super::copy_hbone("hbone client", &mut upgraded, &mut stream).await {
+                            Ok(_) => {
+                                info!("request complete");
+                                Ok(())
+                            }
+                            Err(e) => {
+                                error!("hbone client copy: {}", e);
```

Comment:
shutdown?

Reply:
From the code, shutdwown will not be called when copy error happens: https://github.com/tokio-rs/tokio/blob/master/examples/proxy.rs
And the shutdown will also return another error if called.

Reply:
The above reply should be for "This is changing, no shutdown when err" . ;-)

No need "shutdown" ? For the normal case, I didn't see 'shutdown', I missed it ?